src/App.js

```diff
@@ -1,23 +1,337 @@
+
+  // information sharing between components
+  const [sharedState, setSharedState] = useState({});
+
+  // firebase favourites array/state
+  const [fireStorage, setFireStorage] = useState([]);
+
+  // error state
+  const [error, setError] = useState(false);
+
+
+
+  // make an API call to the Google Books API
+  useEffect(() => {
+    if(searchTerm.length > 0) {
+      axios({
+        url: `https://www.googleapis.com/books/v1/volumes`,
+        method: 'GET',
+        dataResponse: 'json',
+        params: {
+          key: 'AIzaSyB3ql9-1V1P5fP6TqBRukTZGo7Y5WQgjtk',
+          q: searchTerm[0],
+          printType: searchTerm[2],
+          maxResults: 15,
+          startIndex: searchTerm[1],
+          fields: "items(id, volumeInfo(authors, averageRating, title, subtitle, description, imageLinks, infoLink, printType ))"
+        }
+  
+      })
+      .then((response) => {
+        setIsHidden(true)
+        setBooks(response.data.items)
+      })
+      .catch((error) => {
+        setIsHidden(true);
+        setError(true);
+      })
+    }
+  }, [searchTerm])
+
+  // reset the counter when subject is changed
+  const changeCounter = () => {
+    setCounter(10);
+  }
+
+  // initializing random page to some data
+  useEffect(() => {
+    setSearchTerm(["fiction", Math.floor(Math.random() * 200)]);
+  }, [])
+
+
+  const handleSubmit = (event, userSubject, mediumType) => {
+    if(event !== 'click') {
+      event.preventDefault();
+
+    }
+
+      setBooks([]);
+      setIsHidden(false);
+      setView(false);
+
+    // if user clicks again on search, fresh books will be gotten with the offset param in order to repopulate the page
+      setOffset(counter);
+      setSearchTerm([userSubject, offset, mediumType]);
+      setCounter(counter + 15);
+
+  }
+
+  // const handleSubject = (event) => {
+  //   setUserSubject(event.target.value);
+  // }
+
+  const handleAddBook = (e) => {
+      // const copyArr = books.filter((singleBook) => {
+      //   return singleBook.id !== e.target.value;
+      // })
+      // setBooks(copyArr);
+
+      const newArr = books.map((individualBook) => {
+        if (individualBook.id === e.target.value) {
+          individualBook.clicked = true;
+          return individualBook;
+        }
+
+        return individualBook;
+      })
+
+      setBooks(newArr);
+      // check if firebase already has this book in which case dont add it
+      const clickedBookId = e.target.value;
+      const bookObj = {};
+
+      books.forEach((book) => {
+        if(book.id === clickedBookId) {
+          const imgSrc = book.volumeInfo.imageLinks.smallThumbnail;
+          const title = book.volumeInfo.title;
+          
+          bookObj.src = imgSrc;
+          bookObj.title = title;
+
+          return;
+        }
+      })
+
+      // console.log(bookObj)
+
+
+      // loop through our local reference of firebase 
+      // here is a boolean that needs to remain false in order for the book to be added
```

Comment:
This is an effective method, solid.
What do you get if you cross a vampire with a snow man?
Frost bite.

I am so sorry.